Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/fragments/PermissionsFragment.kt

```diff
@@ -31,34 +35,65 @@ private val PERMISSIONS_REQUIRED = arrayOf(
         Manifest.permission.CAMERA,
         Manifest.permission.RECORD_AUDIO)
 
+// TODO: Change fragment name in refactor commit. Waiting for merge of CameraActivity and AnalyserActivity.
 /**
  * This [Fragment] requests permissions and, once granted, it will navigate to the next fragment
+ * with the lowest camera settings available.
  */
 class PermissionsFragment : Fragment() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-
+        
         if (hasPermissions(requireContext())) {
             // If permissions have already been granted, proceed
-            Navigation.findNavController(requireActivity(), R.id.fragment_container).navigate(
-                    PermissionsFragmentDirections.actionPermissionsToSelector())
+            startCamera()
         } else {
             // Request camera-related permissions
             requestPermissions(PERMISSIONS_REQUIRED, PERMISSIONS_REQUEST_CODE)
         }
     }
 
+    private fun startCamera(){
+        val cameraManager =
+                requireContext().getSystemService(Context.CAMERA_SERVICE) as CameraManager
+
+        val cameraList = enumerateHighSpeedCameras(cameraManager)
+
+        val lowestSetting = findLowestCameraSetting(cameraList)
+
+        Navigation.findNavController(requireActivity(), R.id.fragment_container).navigate(
```

Comment:
Any reason why we find the camera options here in the permissions fragment, rather than in the Camera fragment?

Reply:
That makes more sense. As discussed I will move this logic to CameraFragment.